aea/skills/base.py

```diff
@@ -585,6 +587,8 @@ def __init__(
 
         self._contracts = {}  # type: Dict[str, Contract]
 
+        self._skill_context._skill = self
```

Comment:
We now take care of setting the skill context here, making usage more straightforward.